src/__test__/components/data-exploration/generic-gene-table/ExpressionCellSetModal.test.jsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { act } from 'react-dom/test-utils';
+import { Provider } from 'react-redux';
+import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
+import { makeStore } from 'redux/store';
+
+import ExpressionCellSetModal from 'components/data-exploration/generic-gene-table/ExpressionCellSetModal';
+
+import { GENES_SELECT } from 'redux/actionTypes/genes';
+import { dispatchWorkRequest } from 'utils/work/seekWorkResponse';
+import { loadBackendStatus } from 'redux/actions/backendStatus';
+import { updateExperimentInfo } from 'redux/actions/experimentSettings';
+
+import mockAPI, { generateDefaultMockAPIResponses } from '__test__/test-utils/mockAPI';
+import fake from '__test__/test-utils/constants';
+import endUserMessages from 'utils/endUserMessages';
+
+enableFetchMocks();
+
+const mockOnCancel = jest.fn();
+
+jest.mock('utils/work/seekWorkResponse', () => ({
+  seekFromS3: jest.fn(),
+  dispatchWorkRequest: jest.fn(),
+}));
+
+const renderExpressionCellSetModal = async (storeState) => {
+  await act(async () => {
+    render(
+      <Provider store={storeState}>
+        <ExpressionCellSetModal onCancel={mockOnCancel} />
+      </Provider>,
+    );
+  });
+};
+
+let storeState;
+const selectedGenes = ['DOK3', 'DOK4'];
+const experimentId = fake.EXPERIMENT_ID;
+
+const createButtonText = /^Create$/i;
+const loadingText = /Creating cell set.../i;
+
+describe('ExpressionCellSetModal', () => {
+  beforeEach(async () => {
+    jest.clearAllMocks();
+
+    storeState = makeStore();
+
+    fetchMock.resetMocks();
+    fetchMock.mockIf(/.*/, mockAPI(generateDefaultMockAPIResponses(experimentId)));
```

Comment:
what does thi `mockIf` do?

Reply:
This setsup mock for a specific URL. If the requested URL matches the pattern, it'll be passed to the callback function. Our callback function handles all routes, so it's matched against all URLs.